src/Surfnet/AzureMfa/Domain/Institution/Collection/InstitutionCollection.php

```diff
@@ -46,4 +51,18 @@ public function getByName(string $name) : Institution
         }
         return $this->institutions[$name];
     }
+
+    public function getByEmailDomain(EmailAddress $address) : Institution
+    {
+        foreach ($this->institutions as $institution) {
+            /** @var EmailDomainInterface $domain */
+            $domainCollection = $institution->getEmailDomainCollection();
```

Comment:
Isn't this too much responsibility for a Collection?

Reply:
Personally I think the logic should be move to the `EmailDomainMatchingService` as it seem to do just that.